cmake/build_helpers.cmake

```diff
@@ -17,4 +17,39 @@ function(ginkgo_compile_features name)
     if(GINKGO_WITH_IWYU AND GINKGO_IWYU_PATH)
         set_property(TARGET "${name}" PROPERTY CXX_INCLUDE_WHAT_YOU_USE ${GINKGO_IWYU_PATH})
     endif()
+    if(GINKGO_CHANGED_SHARED_LIBRARY)
+        # Put all shared libraries and corresponding imported libraries into the specficed path
+        set_property(TARGET "${name}" PROPERTY
+            RUNTIME_OUTPUT_DIRECTORY "${GINKGO_WINDOWS_SHARED_LIBRARY_PATH}")
+        set_property(TARGET "${name}" PROPERTY
+            ARCHIVE_OUTPUT_DIRECTORY "${GINKGO_WINDOWS_SHARED_LIBRARY_PATH}")
+        if(GINKGO_CHECK_PATH)
+            ginkgo_check_shared_library("${CMAKE_SHARED_LIBRARY_PREFIX}${name}${CMAKE_SHARED_LIBRARY_SUFFIX}")
+        endif()
+    endif()
 endfunction()
+
+function(ginkgo_check_shared_library name)
+    set(PATH_LIST $ENV{PATH})
+    set(PASSED_TEST FALSE)
+    foreach(ITEM ${PATH_LIST})
+        string(REPLACE "\\" "/" ITEM "${ITEM}")
+        if("${ITEM}" STREQUAL "${GINKGO_WINDOWS_SHARED_LIBRARY_PATH}")
+            set(PASSED_TEST TRUE)
+            break()
+        else()
+            # If any path before this build, the path must not contain the ginkgo shared library
+            set(EXISTED_DLL "EXISTED_DLL-NOTFOUND")
```

Comment:
You do not have to set this, CMake takes care of it in the case that the file was not found.
See: https://cmake.org/cmake/help/v3.9/command/find_file.html

Reply:
I set the variable for compiling many times.
If I do not set this variable. it saves the variable in the cache.
Thus, cmake will not check it again after user changes the path.

Reply:
Ah, I didn't think about that. I don't know if that's a CMake bug though. Thanks for the explanation !

Reply:
@tcojean I use `unset(EXISTING_DLL CACHE)` after `find_file` before termination. It also works but does not leave `EXISTING_DLL` in the cache.